dashboard/pages/correlation/callbacks.py

```diff
@@ -187,16 +190,72 @@ def on_visualization_stage_entry(
             full_html=False, include_plotlyjs="cdn"
         ),
     }
-
     return feature_fig, concentration_fig, report_data_correlation_plots, False
 
 
+def on_threshold_change(
+    threshold_1: float,
+    threshold_2: float,
+    stored_uuid: str,
+    file_storage: FileStorage,
+):
+    """
+    Callback for threshold update, updates the plot
+
+    :param threshold_1: first threshold
+    :param threshold_2: second threshold
+    :param stored_uuid: session uuid
+    :param file_storage: file storage
+    :return: figures
+    """
+    saved_name = f"{stored_uuid}_correlation_df.pq"
+    df = pd.read_parquet(pa.BufferReader(file_storage.read_file(saved_name)))
+    feature = "% ACTIVATION" if "% ACTIVATION_0" in df.columns else "% INHIBITION"
+
+    new_fig = concentration_plot(
+        df,
+        feature[2:],
+        threshold_1,
+        threshold_2,
+    )
+
+    df["is_above_threshold_one"] = False
+    df.loc[df[f"{feature}_0"] > threshold_1, "is_above_threshold_one"] = True
+    df["is_above_threshold_two"] = False
+    df.loc[df[f"{feature}_0"] > threshold_2, "is_above_threshold_two"] = True
```

Comment:
I'd rename the columns to: `f"> {threshold_1}"`, `< f"> {threshold_2}"` and  create them in `on_save_filtering_click`  using thresholds' State.